conftest.py

```
@@ -52,18 +77,50 @@ def reset_environment():
     os.environ.setdefault("FEATURE_PREMIUM_NUTRITION", "true")
-    os.environ.setdefault("API_KEY", "test_key")
+    # Don't set API_KEY to enable lenient mode in tests (accepts any non-trivial key)
+    # os.environ.setdefault("API_KEY", "test_key")
     os.environ.setdefault("VIP_MODULE_ENABLED", "true")
     os.environ.setdefault("APP_ENV", "test")
     os.environ.setdefault("ALLOW_DEV_API_KEY", "true")
     os.environ.setdefault("PYTHONPATH", ".:core:app:tests")
 
+    # Override API key validation for all tests
+    try:
+        import app as app_module
```

Flake8 (7.2.0)

[error] 'app as app_module' imported but unused

(F401)